picosha2.h

```diff
@@ -369,7 +373,43 @@ template <typename InContainer>
 std::string hash256_hex_string(const InContainer& src) {
     return hash256_hex_string(src.begin(), src.end());
 }
-
-}  // namespace picosha2
-
+std::string hash256_from_ifstream(std::ifstream f){
+    std::vector<char> data;
+    data.insert(data.begin(),std::istreambuf_iterator<char>(f), std::istreambuf_iterator<char>());
+    std::string hashed;
+    hash256_hex_string(std::string(data.begin(),data.end()), hashed);
+    return hashed;
+}
+inline void multithread(std::vector<std::string> src, std::deque<std::vector<std::string>>* solved){
```

Comment:
FYI: Thread is supported from C++11. And under C++11, lambda is supported. Using lambda function instead named function may avoid users' confusion.